sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldNotBePublic.CSharp9.cs

```diff
@@ -0,0 +1,30 @@
+﻿public record Sample
+{
+    public readonly double Pi3 = 3.14;  // Noncompliant {{Make 'Pi3' private.}}
```

Comment:
can you add a test with 
- static field
- const field

Reply:
Rule's not relevant for static fields. 
https://github.com/SonarSource/sonar-dotnet/blob/204f23b90d8c3e70c1921f01df3877989f8ff5f5/sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/FieldShouldNotBePublicBase.cs#L37

Case with const field is in the `FieldShouldNotBePublic.cs`. For C# 9 context, I wanted to test only the minimal compliant/noncompliant scenarios.

Reply:
regarding the argument that the rule is just for static, if we remove that line, will a test fail?

if we remove lines from a rule and its tests don't fail, are we really testing the behavior of the rule?

Reply:
nevermind, there are tests for static in the main test case